### ISSUE TITLE
Rails 5.1 Support

### ISSUE THREAD

**Issue by atwoodjw** [ActsAsParanoid/acts_as_paranoid#88]
Looks like Rails 5.1 is supported in `master`, but not the most recent release (0.5.0). Can we get a new release?

**Comment by kyleramirez** [comment]
I'm using Rails 5.1 and I still get this warning every time I load the rails environment because of this gem:

    DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from require at /Users/xxx/.rvm/gems/ruby-2.3.0@global/gems/bundler-1.14.6/lib/bundler/runtime.rb:91)

Is this related?

**Comment by jellymann** [comment]
@kyleramirez Yes. If you use the master branch, that deprecation warning goes away. However, a new deprecation warning now appears:

```
DEPRECATION WARNING: Passing a class to the `class_name` is deprecated and will raise an ArgumentError in Rails 5.2. It eagerloads more classes than necessary and potentially creates circular dependencies. Please pass the class name as a string: `belongs_to :primary_category, class_name: 'Category'` (called from belongs_to_with_deleted at /Users/daniel/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/bundler/gems/acts_as_paranoid-6fabadbb31e8/lib/acts_as_paranoid/associations.rb:14)
```

**Comment by etagwerker** [comment]
@zzak Hi, first of all thanks for maintaining this gem! 💯 

Secondly is there an estimate date of release for the next version? 

We're using `acts_as_paranoid` in an app that uses Rails 5.1 and we'd like to depend less on Github (currently using `master`)  and more on Rubygems. 

If I can help make this happen please let me know.

**Comment by patientdev** [comment]
I've been depending on `master` for 5.1 support since June 1.

**Comment by jhubert** [comment]
@zzak Same issue here. A new release would be great as we're doing a 5.1 upgrade. Anything we can do to help?

**Comment by buncismamen** [comment]
have you guys check this project ? https://github.com/rubysherpas/paranoia

it said 

> Paranoia is a re-implementation of acts_as_paranoid for Rails 3/4/5, using much, much, much less code.

**Comment by zzak** [comment]
Hey ya'll, thanks for letting me know as I'm always forgetting about this one :)

Will try to cut something soon! Happy new year!

**Comment by shanecav84** [comment]
*bump*

**Comment by prodeniser** [comment]
I ran into some issues on Rails 5.0 and had to install directly from Github. Would it be possible to get a RubyGems release with the Rails 5.0 and 5.1 fixes?

**Comment by benlangfeld** [comment]
@zzak Any chance you could cut a release, or give someone else permission to do so?

**Comment by KevinColemanInc** [comment]
@benlangfeld / @zzak 

At this point, this should also support rails 5.2

**Comment by nbarthelemy** [comment]
@zzak +1 for @benlangfeld comment about cutting a release. Thanks in advance 👍

**Comment by mvz** [comment]
A new release is available: 0.6.0.
